lib/diagnostic.js

```diff
@@ -17,36 +17,62 @@ const fs = require('fs');
 // resolved or rejected as appropriate
 // use the following as the new function signature
 // const sumLines = (filename) => {
-const sumLines = (filename, callback) => {
-  fs.readFile(filename, { encoding: 'utf8' }, (err, lines) => {
-    let lno = 0;
-    let sum = lines.split('\n').reduce((prev, curr, i) => {
-      lno = i;
-      return prev + (+curr);
-    }, 0);
-    let error = isNaN(sum) && new Error(`line ${lno}: not a number`);
-    callback(error, sum);
-  });
-};
 
-const sumLinesOfArbitraryLengthFile = (filename, callback) => {
-  const rl = require('readline');
-  const rli = rl.createInterface({
-    input: fs.createReadStream(filename),
-  });
-  let lno = 0;
-  let sum = 0;
-  rli.on('line', (line) => {
-    lno++;
-    sum += +line;
-  });
-  rli.on('close', () => {
-    let error = isNaN(sum) && new Error(`line ${lno}: not a number`);
-    callback(error, sum);
-  });
+const sumLines = (filename) => {
+  const promiseSumLines = function (filename, options) {
+    return new Promise( (resolve, reject) => {
+      fs.readFile(filename, options, (error, lines) => {
+        if (error) {
+          reject(error);
+        }
+
+        resolve(lines);
+      });
+    });
+  };
+
+  promiseSumLines(filename, { encoding: 'utf8'})
```

Comment:
`sumLines()` is expected to return a promise.  This would have passed three out of the four tests if this line had been `return promiseSumLines...`;